test/e2e/tests/confirmations/contract-interaction-redesign.spec.js

```diff
@@ -0,0 +1,186 @@
+const {
+  defaultGanacheOptions,
+  openDapp,
+  unlockWallet,
+  WINDOW_TITLES,
+  withFixtures,
+} = require('../../helpers');
+const FixtureBuilder = require('../../fixture-builder');
+
+describe('Confirmation Redesign Contract Interaction Component', function () {
+  if (!process.env.ENABLE_CONFIRMATION_REDESIGN) {
+    return;
+  }
+
+  it(`Sends a contract interaction type 0 transaction without custom nonce editing`, async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesController({
+            preferences: { redesignedConfirmationsEnabled: true },
+          })
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test?.fullTitle(),
```

Comment:
Question: when is the condition when `this.test` is undefined?

Reply:
We only noticed this recently on TypeScript tests, because the type requires the optional chaining. It doesn't make much of a difference in JavaScript files AFAIK.

```js
/**
 * Test context
 *
 * @see https://mochajs.org/api/module-Context.html#~Context
 */
 class Context {
    private _runnable;

    test?: Runnable | undefined;
```